pynicotine/gtkgui/userbrowse.py

```diff
@@ -375,12 +380,7 @@ def __init__(self, userbrowses, user):
             )
         else:
             self.file_popup_menu.add_items(
-                ("#" + _("_Download File(s)"), self.on_download_files),
-                ("#" + _("Download File(s) _To…"), self.on_download_files_to),
-                ("", None),
-                ("#" + _("_Download Folder"), self.on_download_folder),
-                ("#" + _("Download Folder _To…"), self.on_download_folder_to),
-                ("", None),
+                ("#" + _("_Download…"), self.on_download_files),
```

Comment:
The dialog implementation is not required in the `file_popup_menu` because the `userbrowse` interface already provides the needed selection for what files to download.

Instead, consider replacing the "Download Folder & Subfolders _To…" function in `folder_popup_menu`, if it is possible to pass file data recursively from cached shares data, then the dialog would be helpful to preview exactly what will be queued and to choose a download location to download to.

Reply:
> Instead, consider replacing the "Download Folder & Subfolders _To…" function in `folder_popup_menu`, if it is possible to pass file data recursively from cached shares data (while keeping the full path structure), then the dialog would be helpful to preview exactly what will be queued and to choose a download location to download to.

I'll see what I can do. What I was concerned about was the four different download items in the file menu. Adding the download dialog for recursive folder downloads seems like a good solution that would allow us to remove the two folder download items from the file menu.

Reply:
Should be working now.